### ISSUE TITLE
queryBuilder.extract cannot accept namespace bindings

### ISSUE THREAD

**Issue by kcoleman-marklogic** [marklogic/node-client-api#341]
`queryBuilder.extract` thinks you can feed it namespace bindings for use in the extract path expressions. The original spec, the impl, and the JSDoc all assert this. Three cheers for consistency and ease of use.  Alas, this does not, in fact, work. 

When I tried this, the node client returned a 500 (how rude) and the server log contained an XDMP-UNBPRFX error. If I pre-define my namespace prefix on my group using the Admin UI, then the extract worked fine.

This behavior is consistent with the `extract-document-data` option documentation, which says you have to pre-define namespace bindings when using JSON options. The JSON option syntax does not allow for specification of namespace bindings. See the following:

http://docs.marklogic.com/guide/search-dev/appendixa#id_98906

I will leave it to you, Erik & Co. to ponder whether or not we want to address the underlying systemic problem. If we do not make a systemic change, you need to change the impl and the JSDoc for `queryBuilder.extract`, and I need to change its documentation in the Node.js dev guide.

Below is the example code I was trying to run. For purposes of this problem, the content you're querying is irrelevant.
```javascript
const marklogic = require('marklogic');

// MODIFY THIS VAR TO MATCH YOUR ENV
const connInfo = {
    host: 'localhost',
    port: 8000,
    user: 'admin',
    password: 'admin',
    database: 'Documents'
  };
const db = marklogic.createDatabaseClient(connInfo);
const qb = marklogic.queryBuilder;

db.documents.query(
  qb.where(
    qb.collection('person-envelopes'),
    qb.parsedFrom('entity-type:person AND washington',
      qb.parseBindings(
        qb.value(
          qb.element(
            qb.qname('http://marklogic.com/entity-services','title')),
          qb.bind('entity-type'))))
  ).slice(qb.extract({
       paths: ['//es:instance/(Person)'],
       selected: 'include',
       namespaces: {'es': 'http://marklogic.com/entity-services'}
     }))
).result( function(results) {
  for (let result of results) {
    console.log(JSON.stringify(result));
  }
});
```

**Comment by kcoleman-marklogic** [comment]
Just wanted to add one more thing about this bug: The Node.js API does not, AFAIK, expose the /v1/config/namespaces endpoint, so you cannot workaround this problem wholly within the purview of Node.js. 

You must use the REST API directly (or Java, for the masochists among us) to configure the namespace binding and make `qb.extract` work. The REST API will not take advantage of namespace bindings on the group or the db (configured via the Admin interface).

**Comment by ehennum** [comment]
I'm wondering whether the bug here is on the REST API.  Shouldn't the JSON payload support namespace bindings?  Why is it necessary to predeclare the namespaces for JSON and JSON only?

**Comment by ehennum** [comment]
The 52800 internal bug fix to the REST API should appear in 10.0-2 and 9.0-11.

The Node.js API will need to send the namespaces in a structure similar to the following

`{"extract-document-data":{
    "selected":"include", 
    "extract-path":["//prefix1:path1","/prefix2:path2"], 
    "namespaces":{"prefix1":"http://uri1","prefix2":"http://uri2"}}}`

**Comment by georgeajit** [comment]
Test ran fine.
